### PR TITLE
Feature/slls 131 expose context sensitive descriptions

### PR DESCRIPTION
**Pull request by vojtech-suchy-sonarsource**

### CODE REVIEW

File: src/test/java/org/sonarsource/sonarlint/ls/mediumtests/LanguageServerMediumTests.java [hunk 1]
```diff
@@ -896,9 +895,19 @@ void openHotspotInBrowserShouldLogIfWorkspaceNotFound() {
...
 
+  @Test
+  void test_open_hotspot_rule_description() {
+    var FAKE_RULE_KEY = "ruleKey";
+    var FAKE_HOTSPOT_KEY = "hotspotKey";
+    var FAKE_FILE_URI = "file://sample.py";
+    var params = new SonarLintExtendedLanguageServer.ShowHotspotRuleDescriptionParams(FAKE_RULE_KEY, FAKE_HOTSPOT_KEY);
+    params.setFileUri(FAKE_FILE_URI);
+
+    assertThat(lsProxy.showHotspotRuleDescription(params)).isNull();
```
Comment (jblievremont) [inline src/test/java/org/sonarsource/sonarlint/ls/mediumtests/LanguageServerMediumTests.java:906]: There could be a check that the correct parameters (including the context) are passed to the client, e.g using `client.ruleDesc`.

**Review by jblievremont — APPROVED** [review]
See my comment on testing "show hotspot description", otherwise LGTM 👍🏻

File: src/main/java/org/sonarsource/sonarlint/ls/CommandManager.java [hunk 2]
```diff
@@ -278,15 +298,36 @@ private void openRuleDescription(@Nullable String fileUri, String ruleKey) {
...
-    var htmlDescription = getHtmlDescription(ruleDetails);
-    var htmlDescriptionTabs = getHtmlDescriptionTabs(ruleDetails);
+    var htmlDescription = getHtmlDescription(ruleDetails.getDescription());
+    var htmlDescriptionTabs = getHtmlDescriptionTabs(ruleDetails.getDescription(), ruleContextKey);
     var type = ruleDetails.getType();
     var severity = ruleDetails.getSeverity();
     client.showRuleDescription(new ShowRuleDescriptionParams(ruleKey, ruleName, htmlDescription, htmlDescriptionTabs, type, severity, paramDetails));
   }
 
+  private void showStandaloneRuleDescription(String ruleKey, GetStandaloneRuleDescriptionResponse ruleDetails) {
```
Comment (sophio-japharidze-sonarsource) [inline src/main/java/org/sonarsource/sonarlint/ls/CommandManager.java:321]: Introduced a separate method for standalone rules.

File: src/main/java/org/sonarsource/sonarlint/ls/CommandManager.java [hunk 3]
```diff
@@ -278,15 +298,36 @@ private void openRuleDescription(@Nullable String fileUri, String ruleKey) {
       });
   }
 
-  private void showRuleDescription(String ruleKey, ActiveRuleDetailsDto ruleDetails, Collection<ActiveRuleParamDto> paramDetails) {
+  private void openStandaloneRuleDescription(String ruleKey) {
+    backendServiceFacade.getStandaloneRuleDetails(ruleKey)
```
Comment (sophio-japharidze-sonarsource) [inline src/main/java/org/sonarsource/sonarlint/ls/CommandManager.java:302]: this is the new RulesService method

File: src/main/java/org/sonarsource/sonarlint/ls/SonarLintExtendedLanguageClient.java [hunk 4]
```diff
@@ -112,6 +114,19 @@ public ShowRuleDescriptionParams(String ruleKey, String ruleName, String htmlDes
       this.parameters = params.stream().map(p -> new RuleParameter(p.getName(), p.getDescription(), p.getDefaultValue())).toArray(RuleParameter[]::new);
     }
 
+    public ShowRuleDescriptionParams(String ruleKey, String ruleName, String htmlDescription, RuleDescriptionTab[] htmlDescriptionTabs,
+      RuleType type, IssueSeverity severity, Map<String, RuleParamDefinitionDto> params) {
```
Comment (sophio-japharidze-sonarsource) [inline src/main/java/org/sonarsource/sonarlint/ls/SonarLintExtendedLanguageClient.java:118]: new constructor that accepts `RuleParamDefinitionDto` for standalone rules

File: src/main/java/org/sonarsource/sonarlint/ls/backend/BackendService.java [hunk 5]
```diff
@@ -36,32 +38,54 @@
...
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new IllegalStateException("SonarLint backend initialization interrupted", e);
+    }
+    return backend;
   }
 
   public void openHotspotInBrowser(OpenHotspotInBrowserParams params) {
-    backend.getHotspotService().openHotspotInBrowser(params);
+    initializedBackend().getHotspotService().openHotspotInBrowser(params);
```
Comment (sophio-japharidze-sonarsource) [inline src/main/java/org/sonarsource/sonarlint/ls/backend/BackendService.java:81]: we need to make sure that the backend is properly initialized before making any further requests

**Review by jblievremont — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**